trading/views.py

```diff
@@ -0,0 +1,99 @@
+from rest_framework import status, mixins, viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from trading.models import Offer, Trade
+from user.permissions import (
+    BlackListPermission,
+    IsOwner,
+    CreateOfferPermission
+)
+from trading.serializers import (
+    OfferRetrieveSerializer,
+    OfferCreateSerializer,
+    OfferListSerializer,
+    TradeListSerializer,
+    TradeRetrieveSerializer
+)
+from trading.filters import OfferUserFilter
+
+
+class OfferViewSet(mixins.ListModelMixin,
+                   mixins.CreateModelMixin,
+                   mixins.RetrieveModelMixin,
+                   viewsets.GenericViewSet):
+    """A viewset for model Offer that provides `retrieve`,
+     `create`, and `list` actions with specific permissions by action."""
+    model = Offer
+    queryset = Offer.objects.all()
+    filterset_class = OfferUserFilter
+    permission_classes = (IsAuthenticated, BlackListPermission)
```

Comment:
i think you do not need this declaration according to `permission_classes_by_action` below. And also first always goes variables and only after them goes methods. DO NOT MIX THEM